spec/enumerable_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require_relative '../my_enumerable_module.rb'
+
+# rubocop: disable Metrics/BlockLength, Layout/LineLength
+
+describe Enumerable do
+  arr = [5, 6, 7]
+  describe '#my_each' do
+    it 'Returns the same array if block is given' do
+      expect(arr.my_each { |x| x }).to eql(arr)
+    end
+    it 'Returns an enumerable if block is not given' do
+      expect(arr.my_each.class).to eql(Enumerator)
+    end
+  end
+  describe '#my_each_with_index' do
+    it 'Returns the same array if block is given' do
+      expect(arr.my_each_with_index { |x, y| x + y }).to eql(arr)
+    end
+    it 'Returns an enumerable if block is not given' do
+      expect(arr.my_each_with_index.class).to eql(Enumerator)
+    end
+  end
+  describe '#my_select' do
+    it 'Returns result of the condition inside the block if block is given' do
+      expect(arr.my_select { |x| x > 5 }).to eql([6, 7])
+    end
+    it 'Returns an enumerable if block is not given' do
+      expect(arr.my_select.class).to eql(Enumerator)
+    end
+  end
+  describe '#my_all?' do
```

Comment:
* [ ] ```my_all``` has intested functionality. 
What happen when dealing with regular expressions?

Reply:
Hi Forison

It's included in this test case:
```   
it 'Returns true if all arrays values satisfy the condition of the given argument' do
  expect(arr.my_all?(Integer)).to eql(true)
end
```
if the argument is a regex, it will execute it and return true if all the elements are targeted.

I added it as a test case anyways...